apps/daimo-mobile/src/sync/syncLandline.ts

```diff
@@ -0,0 +1,86 @@
+import {
+  TransferClog,
+  TransferSwapClog,
+  getTransferClogType,
+} from "@daimo/common";
+
+/**
+ * All old landline clogs should be bundled into a single clog with the most
+ * up-to-date offchainTransfer.
+ *
+ * Landline TransferClog sync strategy:
+ *  1. If an old log matches by tx hash, then it is the on-chain counterpart
+ *     of the incoming landline clog. Keep the on-chain part of the clog and
+ *     update the offchainTransfer to the incoming landline clog's.
+ *  2. If an old log matches by just the transferID, then it is a potentially
+ *     outdated landline clog. Replace the old clog with the incoming landline clog.
```

Comment:
this is some substantially tricky logic, but lgtm.

in what situation would a landline transfer match more than one onchain transfer?

Reply:
it should never have to match more than one onchain transfer, but there is the case where there is an old landline transfer clog and an on-chain transfer in the history that need to get matched

example with user making a deposit:
1. api server syncs the deposit from landline (without a tx hash) and updates the mobile app. mobile app saves this clog in `recentTransfers`
2. onchain transfer from bridge to daimo account happens. mobile app saves this clog in `recentTransfers`
3. bridge webhook fires and api syncs the deposit, which now includes a tx hash. this new clog needs to bundle the clogs in steps 1 and 2 into a single item. it matches clog 1 on `transferId` and clog 2 on `txHash`